### PR TITLE
kie-issues#1498: Add "Generate form code for human interactions" command in the BPMN VS Code extension

### LINKED ISSUE #1498: Add "Generate form code for human interactions" command in the BPMN VS Code extension
Add the form generation command to the BPMN VS Code extension using the "jbpm-form-generator" package introduced in  https://github.com/apache/incubator-kie-issues/issues/1493. The command should be named "Generate form code for human interactions". It will use the compiled JSON Schemas to generate the form code into the "src/resources/forms" folder.

The command should be interactive, giving the user the option to generate a Patternfly or Bootstrap code. Also, it should give the option to select which human interactions are desired.

### PR DESCRIPTION
**Pull request by ljmotta**
Closes: https://github.com/apache/incubator-kie-issues/issues/1498

## Description
- Add `BPMN Editor: Generate form code for human interactions from jBPM project` command.
- Add `BPMN Editor` prefix to the other BPMN Editor VS Code commads.
- Rename `inputSanizationUtil` to `removeInvalidVarChars`.
- Rename `sanatizedId` to `idWithoutInvalidVarChars`.
- Make `theme` `idWithoutInvalidVarChars` optional.
- Improve `form-code-generator` `FormAsset` interface.

### CODE REVIEW

File: packages/bpmn-vscode-extension/package.json [hunk 1]
```diff
@@ -65,11 +69,15 @@
           "dark": "./static/svg-icon-dark.png",
           "light": "./static/svg-icon-light.png"
         },
-        "title": "Get BPMN Editor Preview SVG"
+        "title": "BPMN Editor: Get BPMN Editor Preview SVG"
```
Comment (tiagobento) [inline packages/bpmn-vscode-extension/package.json]:
```suggestion
        "title": "BPMN Editor: Get preview SVG"
```

?

File: packages/bpmn-vscode-extension/src/extension/extension.ts [hunk 2]
```diff
@@ -49,6 +50,10 @@ export function activate(context: vscode.ExtensionContext) {
     backendProxy: backendProxy,
   });
 
+  context.subscriptions.push(
+    vscode.commands.registerCommand("extension.kie.generateFormCode", async (args: any) => generateFormsCommand())
+  );
```
Comment (tiagobento) [inline packages/bpmn-vscode-extension/src/extension/extension.ts]:
```suggestion
  context.subscriptions.push(
    vscode.commands.registerCommand("extension.apache.kie.bpmnEditor.generateFormCode", async (args: any) => generateFormsCommand())
  );
```

?

**Review by tiagobento — CHANGES_REQUESTED** [review]
Few comments inline... and also I think we need to add this command to our development VS Code Extension...

**Review by tiagobento — COMMENTED** [review]
"Invalid var chars" seems like a very clear name, but I'm not sure in what context "var" falls into. Is it a BPMN variable? Or a TypeScript variable? Maybe we could be a little bit extra there, if that's the goal?

**Comment by ljmotta** [comment]
@tiagobento It's related to TypeScript variables... I'll update accordingly.

File: packages/kie-editors-dev-vscode-extension/package.json [hunk 3]
```diff
@@ -96,15 +97,19 @@
...
+        "title": "Kogito Editors: Run"
       },
       {
         "command": "extension.kogito.silentlyGenerateSvg",
-        "title": "Generate SVG without any notification"
+        "title": "Kogito Editors: Generate SVG without any notification"
+      },
+      {
+        "command": "extension.apache.kie.kogitoEditors.generateFormCode",
+        "title": "Kogito Editors: Generate form code for human interactions from jBPM project"
```
Comment (jomarko) [inline packages/kie-editors-dev-vscode-extension/package.json:112]: not sure about word `Kogito` as we are in `kie-editors-dev-vscode-extension` package, not a `kogito-editors-dev-vscode-extension`

File: packages/kie-editors-dev-vscode-extension/package.json [hunk 4]
```diff
@@ -144,7 +149,7 @@
     },
     "customEditors": [
       {
-        "displayName": "KIE Kogito Editors",
+        "displayName": "Apache KIE Kogito Editors",
```
Comment (jomarko) [inline packages/kie-editors-dev-vscode-extension/package.json:152]: similar comment

File: packages/form-code-generator-vscode-command/src/generateFormCodeCommand.ts [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+  const defaultPath = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders[0].uri.fsPath : undefined;
+
+  // Select project path
+  const projectUri = await vscode.window.showOpenDialog({
+    canSelectFiles: false,
+    canSelectFolders: true,
+    canSelectMany: false,
+    openLabel: "Select Project Folder",
+    defaultUri: defaultPath ? vscode.Uri.file(defaultPath) : undefined,
+  });
```
Comment (jomarko) [inline packages/form-code-generator-vscode-command/src/generateFormCodeCommand.ts]: The problem with added dialogues/prompts/popups, not sure what is the terminology in VSCode is only with the first one. There is no additionl linformation in the step of selecting path to the jBPM project. See the second screenshot below.:
### 01
Screenshot 2024-10-22 150300

### 02
Screenshot 2024-10-22 150309

it seems `openLabel: "Select Project Folder",` config is not used? In the documentation I see alternative `title` ? https://code.visualstudio.com/api/references/vscode-api#OpenDialogOptions


Using the generate command for the first time I was confused what should I do in the step **02**.

**Review by jomarko — COMMENTED** [review]
I think the only feedback I have is about the step of selecting jBPM project path

File: packages/form-code-generator-vscode-command/src/generateFormCodeCommand.ts (hunk 5 above)
Comment (ljmotta) [reply packages/form-code-generator-vscode-command/src/generateFormCodeCommand.ts]: Indeed. Changing to title fixed the problem, adding the label in this step:

image

**Comment by ljmotta** [comment]
@jomarko Thanks for your review. The name "Kogito Editors" is due to the current extension name. It's not the intent of this PR to change the extension name, I've only added the "Apache" as it's a must. @tiagobento do we create a new issue to tackle the Kogito name, or I just remove and call it "Apache KIE Editors"?

File: packages/form-code-generator-vscode-command/src/generateFormCodeCommand.ts (hunk 5 above)
Comment (jomarko) [reply packages/form-code-generator-vscode-command/src/generateFormCodeCommand.ts]: perfect

**Review by jomarko — APPROVED** [review]
Thank you for changes, I understand the substring "Kogito Editors" must be discussed further with others, so I am approving this PR and I will accept any decision that will be taken by the team around this.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**